backend/fms_core/viewsets/_fetch_data.py

```diff
@@ -419,32 +440,32 @@ def fetch_export_metadata(self, ids: List[int] =[]) -> Tuple[List, int]:
             if biosample in metadata_per_biosample.keys():
                 metadata_per_biosample[biosample].append(metadata)
             else:
-                metadata_per_biosample[biosample] = [metadata] 
+                metadata_per_biosample[biosample] = [metadata]
 
         serialized_data = []
         if not samples_by_derived:
             serialized_data.append({}) # Allow the returned csv file to be named instead of random name.
         # For the renderer context
         metadata_names = []
-        for sample in samples_by_derived.values():
-            derived_sample = derived_samples[sample['derived_samples']]
-            # Prevents crashing if sample has no metadata
-            biosample_id = derived_sample["biosample__id"]
-            metadata = metadata_per_biosample[biosample_id] if biosample_id in metadata_per_biosample else None
-                
-            data = {
-                'alias': derived_sample["biosample__alias"],
-                'biosample_id': biosample_id ,  
-                'sample_name': sample["name"],   
-                'container_name': sample["container__name"],
-                'container_barcode': sample["container__barcode"],
-                'coordinates': sample["coordinate__name"], 
-                'project': derived_sample["project__name"],
-                **(dict((item["name"], item["value"]) for item in metadata) if metadata else dict())
-            }
-            
-            metadata_names.extend([data_key for data_key in data.keys() if data_key not in metadata_names])
-            serialized_data.append(data)
+        for derived_sample_id, samples in samples_by_derived.items():
+            for sample in samples:
+                derived_sample = derived_samples[derived_sample_id]
+                # Prevents crashing if sample has no metadata
+                biosample_id = derived_sample["biosample__id"]
+                metadata = metadata_per_biosample[biosample_id] if biosample_id in metadata_per_biosample else None
+                data = {
+                    'alias': derived_sample["biosample__alias"],
+                    'biosample_id': biosample_id ,
+                    'sample_name': sample["name"],
+                    'container_name': sample["container__name"],
+                    'container_barcode': sample["container__barcode"],
+                    'coordinates': sample["coordinate__name"],
+                    'project': derived_by_samples[derived_sample_id][sample["id"]],
```

Comment:
we need a better name for this dict than `derived_by_samples`. Maybe projectByDerivedBySampleIdAndSampleId?